docs/latest/design/ratelimit.md

```diff
@@ -0,0 +1,270 @@
+* A `match` holds true, when all conditions under the `match` hold true.
+
+Here's an example highlighting this -
+
+```
+apiVersion: gateway.envoyproxy.io/v1alpha1
+kind: RateLimitFilter
+metadata:
+  name: ratelimit-all-safeguard-app 
+spec:
+  type: Global
+  global:
+    rules:
+    - limit:
+        requests: 100
+        unit: Second
+---
+
+apiVersion: gateway.envoyproxy.io/v1alpha1
+kind: RateLimitFilter
+metadata:
+  name: ratelimit-per-user
+spec:
+  type: Global
+  global:
+    rules:
+    - matches:
+      - headers:
+        - type: Distinct
+          name: x-user-id
+      limit:
+        requests: 1000
+        unit: Hour
+---
+apiVersion: gateway.networking.k8s.io/v1beta1
+kind: HTTPRoute
+metadata:
+  name: example
+spec:
+  parentRefs:
+  - name: eg
+  hostnames:
+  - www.example.com
+  rules:
+  - matches:
+    - path:
+        type: PathPrefix
+        value: /foo
+    filters:
+    - type: ExtensionRef
+      extensionRef:
+        group: gateway.envoyproxy.io
+        kind: RateLimitFilter
+        name: ratelimit-per-user
+    - type: ExtensionRef
+      extensionRef:
+        group: gateway.envoyproxy.io
+        kind: RateLimitFilter
+        name: ratelimit-all-safeguard-app
+    backendRefs:
+    - name: backend
+      port: 3000
+```
+
+* The user has created two `RateLimitFilter`s  and has attached it to a `HTTPRoute` - one(`ratelimit-all-safeguard-app`) to
+ensure that the backend does not get overwhelmed with requests, any excess requests are rate limited irrespective of
+the attributes within the traffic flow, and another(`ratelimit-per-user`) to rate limit each distinct user client
+who can be differentiated using the `x-user-id` header, to ensure that each client does not make exessive requests to the backend.
+* If user `baz` (identified with the header and value of `x-user-id: baz`) sends 90 requests within the first second, and
+user `bar` sends 11 more requests during that same interval of 1 second, and user `bar` sends the 101th request within that second,
+the rule defined in `ratelimit-all-safeguard-app` gets activated and Envoy Gateway will ratelimit the request sent by `bar` (and any other
+request sent within that 1 second). After 1 second, the rate limit counter associated with the `ratelimit-all-safeguard-app` rule
+is reset and again evaluated.
+* If user `bar` also ends up sending 90 more requests within the hour, summing up `bar`'s total request count to 101, the rate limit rule
+defined within `ratelimit-per-user` will get activated, and `bar`'s requests will be rate limited again until the hour interval ends.
+* Within the same above hour, if `baz` sends 991 more requests, summing up `baz`'s total request count to 1001, the rate limit rule defined
+within `ratelimit-per-user` will get activated for `baz`, and `baz`'s requests will also be rate limited until the hour interval ends. 
+
+## Design Decisions
```

Comment:
worth touching on what the ordering between adding this filter and the JWT or other authz filters will be, is it dependent on user configuration and the order of filters you set on a route rule?

Reply:
since this is a user facing design doc, prefer adding such info about envoy filter ordering in later design PRs

Reply:
i think this is a user concern, users may want to know how to specify that their RL is applied before authz to protect their auth service or vice versa

Reply:
I think ordering should be added in Authn design docs to highlight precedence over other filters, RL should be appended in a default way, will track this using a GH Issue, to ensure this is part of https://github.com/envoyproxy/gateway/issues/336

Reply:
> will track this using a GH Issue, to ensure this is part of #336

@arkodg can you create a tracking issue so this feedback is not lost when the PR is merged?

Reply:
raised https://github.com/envoyproxy/gateway/issues/847